lib/auth/auth.go

```diff
@@ -5165,17 +5165,39 @@ func (a *Server) ExportUpgradeWindows(ctx context.Context, req proto.ExportUpgra
 	return rsp, nil
 }
 
-func (a *Server) isMFARequired(ctx context.Context, checker services.AccessChecker, req *proto.IsMFARequiredRequest) (*proto.IsMFARequiredResponse, error) {
+// MFARequiredToBool translates a [proto.MFARequired] value to a simple
+// "required bool".
+func MFARequiredToBool(m proto.MFARequired) (required bool) {
+	switch m {
+	case proto.MFARequired_MFA_REQUIRED_NO:
+		return false
+	default: // _UNSPECIFIED or _YES are both treated as required.
+		return true
+	}
+}
+
+func (a *Server) isMFARequired(ctx context.Context, checker services.AccessChecker, req *proto.IsMFARequiredRequest) (resp *proto.IsMFARequiredResponse, err error) {
+	// Assign Required as a function of MFARequired.
+	defer func() {
```

Comment:
It feels perhaps a little unintuitive to do this as a `defer` - I could see myself coming to this function which is already fairly long and missing the defer and being confused about why it behaves the way it does.

What do you think ?

Reply:
The idea behind the defer was to make sure that Required is always a function of MFARequired. I can move to manual assignments if y'all think that's best.

Reply:
This suggests to me unnecessary redundancy. What if we deprecate the `Required` field and do something like this (in `api/client/proto/types.go`)?

```go
func (m *IsMFARequiredResponse) IsRequired() bool {
	switch m.MFARequired {
	case MFARequired_MFA_REQUIRED_NO:
		return false
	default: // _UNSPECIFIED or _YES are both treated as required.
		return true
	}
}
```

Reply:
There is some duplication here, although it didn't seem to me like a clear deprecation scenario - some places do want a boolean-like decision, while others care about _UNSPECIFIED. I try to pick my battles when tangling with these parts, so it that's OK I'd rather not pursue this particular tangent right now.

Reply:
Discussed out-of-band, I'll take a TODO with the intent to follow up after my next few PRs. Meanwhile I'll keep the defer. Thanks both and let me know you'd rather see more changes.